h2/src/main/org/h2/mvstore/cache/CacheLongKeyLIRS.java

```diff
@@ -1074,7 +1137,17 @@ private void removeFromQueue(Entry<V> e) {
         void setMaxMemory(long maxMemory) {
             this.maxMemory = maxMemory;
         }
-
+        
+        
+        <T> T withLock(Supplier<T> c) {
+            l.lock();
+            try {
+                return c.get();
+            }
+            finally {
+                l.unlock();
+            }
+        }
```

Comment:
From my point of view this method doesn't make code more readable and introduction of lambda functions can make it slower. Usually it isn't critical, but it looks strange when you're trying to improve performance.